src/wallet/crypter.cpp

```diff
@@ -184,6 +203,21 @@ bool CCryptoKeyStore::Unlock(const CKeyingMaterial& vMasterKeyIn)
             if (fDecryptionThoroughlyChecked)
                 break;
         }
+        CryptedSpendingKeyMap::const_iterator skmi = mapCryptedSpendingKeys.begin();
+        for (; skmi != mapCryptedSpendingKeys.end(); ++skmi)
+        {
+            const libzcash::PaymentAddress &address = (*skmi).first;
+            const std::vector<unsigned char> &vchCryptedSecret = (*skmi).second;
+            libzcash::SpendingKey key;
+            if (!DecryptSpendingKey(vMasterKeyIn, vchCryptedSecret, address, key))
+            {
+                keyFail = true;
+                break;
+            }
+            keyPass = true;
+            if (fDecryptionThoroughlyChecked)
+                break;
+        }
```

Comment:
This made me wonder whether there's a way to have less duplicated code between transparent and Zcash key objects. I couldn't immediately think of any way that wasn't very intrusive to the upstream code.